### PR TITLE
Expose `ResourceFileElements` initializer

### PR DESCRIPTION
**Pull request by kwridan**
### Short description 📝

- To support cases where resources are defined in helpers, we need to support both literal as well as non literal declarations for resource file elements
- Updating fixture to help capture the scenario where non literal declarations are used

### Test Plan 🛠

- Run `swift build & swift run tuist generate --path fixtures/ios_app_with_framework_and_resources`
- Verify generation succeeds

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://tuist.io/docs/contribution/testing-strategy/).
- [ ] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- ~In case the PR introduces changes that affect users, the documentation has been updated.~

### CODE REVIEW

**Comment by kwridan** [comment]
@laxmorek

In your case would making the helper return `ResourceFileElements` instead of `[ResourceFileElement]` fix the issue you ran into?

**Review by fortmarek — APPROVED** [review]
This is actually needed by our company as 1.35.0 turns out to be breaking for our helpers.

Is there something I can do to help to push this, @kwridan?

**Comment by kwridan** [comment]
Thanks @fortmarek - it's good to land. Will push a changelog update

File: fixtures/ios_app_with_framework_and_resources/Framework1/Project.swift [hunk 1]
```diff
@@ -1,13 +1,24 @@
 import ProjectDescription
 
+// Example of resources specified via variables instead of direct literals.
+// Often sources and resources are declared in helpers where their values
+// are computed, as such we need to support non-literal declarations.
+let resourcesDirectory = "Resources"
+let resources: [ResourceFileElement] = [
+    "\(resourcesDirectory)/framework_resource.txt"
+]
```
Comment (laxmorek) [inline fixtures/ios_app_with_framework_and_resources/Framework1/Project.swift:9]: thx for updating the fixture with this use-case :)

**Review by laxmorek — APPROVED** [review]
🎉

**Pull request merged**